### PR TITLE
feat: omit custom fields in subgraph errors

### PR DESCRIPTION
**Pull request by thisisnithin**

### CODE REVIEW

File: v2/pkg/engine/resolve/loader.go [hunk 1]
```diff
@@ -813,6 +816,29 @@ func (l *Loader) optionallyOmitErrorExtensions(values []*astjson.Value) {
...
+	if !l.omitCustomSubgraphErrorFields {
+		return
+	}
+	for _, value := range values {
+		if value.Type() == astjson.TypeObject {
+			obj := value.GetObject()
+			var keysToDelete []string
+			obj.Visit(func(k []byte, v *astjson.Value) {
+				key := unsafebytes.BytesToString(k)
+				if key != "message" && key != "extensions" {
```
Comment (jensneuse) [inline v2/pkg/engine/resolve/loader.go:830]: what if
```json
{"extensions":{"foo":"bar"}}
```

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: v2/pkg/engine/resolve/loader.go (hunk 1 above)
Comment (jensneuse) [reply v2/pkg/engine/resolve/loader.go:830]: Wouldn't it make more sense to explicitly whitelist?

Comment (thisisnithin) [reply v2/pkg/engine/resolve/loader.go:830]: This is handled with `allowed_extension_fields`

**Review by jensneuse — APPROVED** [review]

File: v2/pkg/engine/resolve/loader.go [hunk 2]
```diff
@@ -813,6 +816,29 @@ func (l *Loader) optionallyOmitErrorExtensions(values []*astjson.Value) {
...
+					keysToDelete = append(keysToDelete, key)
+				}
+			})
+			for _, key := range keysToDelete {
+				fmt.Println(key)
```
Comment (StarpTech) [inline v2/pkg/engine/resolve/loader.go:835]: @thisisnithin was this merged? :smile:

**Pull request merged**